pkgs/applications/science/logic/boolector/default.nix

```diff
@@ -40,22 +38,61 @@ stdenv.mkDerivation rec {
     cd ../src
     find . -iname '*.h' -exec cp --parents '{}' $dev/include \;
     rm -rf $dev/include/tests
-  '';
+  '' + (
+    if stdenv.isDarwin then ''
+      install_name_tool -change libboolector.dylib $lib/lib/libboolector.dylib $out/bin/boolector
+      install_name_tool -change libbtor2parser.dylib ${btor2tools.lib}/lib/libbtor2parser.dylib $out/bin/boolector
+      install_name_tool -change libbtor2parser.dylib ${btor2tools.lib}/lib/libbtor2parser.dylib $lib/lib/libboolector.dylib
+      ''
+    else ""
+  );
 
   checkInputs = [ python3 ];
   doCheck = true;
-  preCheck = ''
-    export LD_LIBRARY_PATH=$(readlink -f lib)
+  preCheck =
+    if stdenv.isDarwin then preCheckDarwin
+    else ''
+      export LD_LIBRARY_PATH=$(readlink -f lib)
+      patchShebangs ..
+    '';
+  postCheck =
+    if stdenv.isDarwin then postCheckDarwin
+    else "";
+  preCheckDarwin = ''
+    # back those up since we modify dylibs for tests
+    cp -r lib lib.back
+    cp -r bin bin.back
+
+    install_name_tool -change libgtest.dylib `pwd`/lib/libgtest.dylib lib/libgtest_main.dylib
```

Comment:
Does the build set the install_name correctly in the final library? On darwin `DYLD_LIBRARY_PATH` can be used instead of patching stuff, similar to how that's currently done for the linux build.